core/src/test/java/com/sequenceiq/cloudbreak/service/image/CachedImageCatalogWrapperProviderTest.java

```diff
@@ -0,0 +1,334 @@
+
+        assertEquals(mapToUuid(expectedCatalog.getImages().getBaseImages()), mapToUuid(actualCatalog.getImages().getBaseImages()));
+        assertEquals(mapToUuid(Collections.emptyList()), mapToUuid(actualCatalog.getImages().getCdhImages()));
+        assertEquals(1, expectedCatalog.getImages().getCdhImages().size());
+    }
+
+    @Test
+    public void testFreeipaImageCatalogFilterToRedhat7() throws CloudbreakImageCatalogException, IOException {
+        String path = getPath(FREEIPA_IMAGE_CATALOG_V3_JSON);
+        ReflectionTestUtils.setField(underTest, "etcConfigDir", path);
+        ReflectionTestUtils.setField(underTest, "enabledLinuxTypes", CB_REDHAT7_FILTER);
+
+        CloudbreakImageCatalogV3 actualCatalog = underTest.getImageCatalogWrapper(FREEIPA_IMAGE_CATALOG_V3_JSON).getImageCatalog();
+
+        assertEquals(mapToUuid(Collections.emptyList()), mapToUuid(actualCatalog.getImages().getBaseImages()));
+        assertEquals(mapToUuid(Collections.emptyList()), mapToUuid(actualCatalog.getImages().getCdhImages()));
+        assertEquals(1, actualCatalog.getImages().getFreeIpaImages().size());
+        assertEquals("81851893-8340-411d-afb7-e1b55107fb10", actualCatalog.getImages().getFreeIpaImages().get(0).getUuid());
+    }
+
+    private List<String> getImageCatalogOses(CloudbreakImageCatalogV3 actualCatalog) {
+        return actualCatalog.getImages().getBaseImages().stream()
+                .map(Image::getOs)
+                .distinct()
+                .collect(Collectors.toList());
+    }
+
+    @Test
+    public void testHttpImageCatalogValid() throws CloudbreakImageCatalogException, IOException {
+        String path = getPath(CB_IMAGE_CATALOG_VALID_JSON);
+        String catalogUrl = "http";
+
+        when(clientMock.target(catalogUrl)).thenReturn(webTargetMock);
+        when(webTargetMock.request()).thenReturn(builderMock);
+        when(builderMock.get()).thenReturn(responseMock);
+        when(responseMock.getStatusInfo()).thenReturn(statusTypeMock);
+        when(statusTypeMock.getFamily()).thenReturn(Response.Status.Family.SUCCESSFUL);
+        when(responseMock.readEntity(String.class)).thenReturn(Files.readString(Paths.get(path, CB_IMAGE_CATALOG_V2_JSON)));
+
+        ReflectionTestUtils.setField(underTest, "etcConfigDir", path);
+        ReflectionTestUtils.setField(underTest, "enabledLinuxTypes", Collections.emptyList());
+
+        CloudbreakImageCatalogV3 actualCatalog = underTest.getImageCatalogWrapper(catalogUrl).getImageCatalog();
+
+        CloudbreakImageCatalogV3 expectedCatalog = objectMapper.readValue(Paths.get(path, CB_IMAGE_CATALOG_V2_JSON).toFile(), CloudbreakImageCatalogV3.class);
+
+        assertEquals(mapToUuid(expectedCatalog.getImages().getBaseImages()), mapToUuid(actualCatalog.getImages().getBaseImages()));
+        assertEquals(mapToUuid(expectedCatalog.getImages().getCdhImages()), mapToUuid(actualCatalog.getImages().getCdhImages()));
+
+    }
+
```

Comment:
I think you missed the `@Test` annotation from the `testHttpImageCatalogNotValidJson` method.